### PR TITLE
Update Gradle stuff to 2024 and fix macos library loading issues

### PR DESCRIPTION
**Pull request by srimanachanta**
based on #972

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,7 @@
 distributionBase=GRADLE_USER_HOME
-distributionPath=wrapper/dists
-distributionUrl=https\://services\.gradle\.org/distributions/gradle-8\.3-bin\.zip
+distributionPath=permwrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
+networkTimeout=10000
+validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
+zipStorePath=permwrapper/dists
```
Comment (mcm001) [inline gradle/wrapper/gradle-wrapper.properties:7]: Why change?

File: photon-core/build.gradle [hunk 2]
```diff
@@ -36,24 +36,3 @@ task writeCurrentVersionJava {
 }
 
 build.dependsOn writeCurrentVersionJava
-
```
Comment (mcm001) [inline photon-core/build.gradle:39]: Iirc this code block is require -- why are we removing it

File: photon-lib/build.gradle [hunk 3]
```diff
@@ -16,20 +16,14 @@ test {
...
-wpilibTools.deps.wpilibVersion = wpilibVersion
+wpilibTools.deps.wpilibVersion = wpi.versions.wpilibVersion.get()
 println("Building for WPILib ${wpilibTools.deps.wpilibVersion}")
 
-// From wpilib shared/config.gradle:
-// NativeUtils adds the following OpenCV warning suppression for Linux, but not
-// for macOS
-// https://github.com/opencv/opencv/issues/20269
-nativeUtils.platformConfigs.osxuniversal.cppCompiler.args.add("-Wno-deprecated-anon-enum-enum-conversion")
-nativeUtils.platformConfigs.linuxathena.cppCompiler.args.add("-Wno-deprecated-anon-enum-enum-conversion")
```
Comment (mcm001) [inline photon-lib/build.gradle:30]: Why yeet

File: photon-server/build.gradle [hunk 4]
```diff
@@ -15,68 +15,46 @@ version versionString + (project.hasProperty('pionly') ? "-raspi" : "")
...
 apply from: "${rootDir}/shared/common.gradle"
 
-def nativeConfigName = 'wpilibTestNative'
-def nativeConfig = configurations.create(nativeConfigName)
-
-def nativeTasks = wpilibTools.createExtractionTasks { configurationName = nativeConfigName }
-
-nativeTasks.addToSourceSetResources(sourceSets.main)
-
-nativeConfig.dependencies.add wpilibTools.deps.cscore()
```
Comment (mcm001) [inline photon-server/build.gradle:25]: Why remove this code block?

File: photon-server/build.gradle [hunk 5]
```diff
@@ -146,15 +121,7 @@ run {
...
-// task overrideToPi {
-//     doLast {
-//         project.setProperty('ArchOverride', 'linuxarm32')
-//     }
-// }
-
-
-
-task deploy {
+task deployIdc {
```
Comment (mcm001) [inline photon-server/build.gradle:124]: Que

Comment (srimanachanta) [reply photon-server/build.gradle:124]: This conflicts with GradleRIO's deploy task, we need to rename this to something better

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (srimanachanta) [reply gradle/wrapper/gradle-wrapper.properties:7]: I kept getting a gradle error on bump and doing `./gradlew wrapper` resolved it and did this. Not sure if it is needed. The standalone project does [it](https://github.com/wpilibsuite/StandaloneAppSamples/blob/main/Java/gradle/wrapper/gradle-wrapper.properties) to.

Comment (mcm001) [reply gradle/wrapper/gradle-wrapper.properties:7]: lol, just curious

File: photon-server/build.gradle (hunk 5 above)
Comment (mcm001) [reply photon-server/build.gradle:124]: Why are we using gradlerio in photon code?

Comment (srimanachanta) [reply photon-server/build.gradle:124]: Not sure, the standalone example did so to track dependency versions, might be a wrapper in native utils, not sure

Comment (mcm001) [reply photon-server/build.gradle:124]: we do need the deploy task to work before I ship this, either option is good w me

**Comment by mcm001** [comment]
The JAR that Actions spits out runs on Windows for me at least

**Review by mcm001 — APPROVED** [review]

**Pull request merged**